packages/typespec-ts/test/modularUnit/scenarios/samples/client/disableHierarchyClient.md

```diff
@@ -0,0 +1,159 @@
+# Should generate samples for disabled hierarchy client
+
+Sample generation should handle hierarchy client successfully.
+
+## TypeSpec
+
+This is tsp definition.
+
+```tsp
+model A {
+  prop1: string;
+}
+
+@route("b")
+namespace B {
+  @route("e")
+  namespace E {
+    model A {
+      prop3: string;
+    }
+    interface C {
+      op opBEC(@body body: A): void
+    }
+  }
+  op opB(@body body: A): void;
+}
+
+@route("/d")
+interface D {
+  @doc("show example opD")
+  op opD(@body body: A): void;
+}
+@doc("show example opTopLevel")
+op opTopLevel(@body body: A): void;
```

Comment:
can you add a case where the operation names are the same but in different namespace?

Reply:
sure, let me add one.

Reply:
issue tracked here: https://github.com/Azure/autorest.typescript/issues/2800